features/netsocket/SocketAddress.cpp

```diff
@@ -122,6 +122,7 @@ static void ipv6_from_address(uint8_t *bytes, const char *addr)
     // Move suffix to end
     memmove(&shorts[NSAPI_IPv6_BYTES/2-suffix], &shorts[0],
             suffix*sizeof(uint16_t));
+    memset(&shorts[0], 0, suffix*sizeof(uint16_t));
```

Comment:
This memset solution not more expensive (slower your writing 2x bytes instead of 1x words (uint16_t) ). The variable `shorts `is already a uint16_t.